### PR TITLE
Connection cleanup

### PR DESCRIPTION
**Pull request by shailesh33**
* Move static internal functions to dyn_connection_internal.c
* Make object printing more generic
* reduce 4 conn_get_* apis into one conn_get
* Move cleanup_charptr to a .c and reduce warnings by 50

### CODE REVIEW

File: src/dyn_connection_internal.c [hunk 1]
```diff
@@ -0,0 +1,256 @@
+/*
+ * Dynomite - A thin, distributed replication layer for multi non-distributed storages.
+ * Copyright (C) 2014 Netflix, Inc.
+ */ 
+
+/*
+ * twemproxy - A fast and lightweight proxy for memcached protocol.
```
Comment (ipapapa) [inline src/dyn_connection_internal.c]: This is a new function that does not exist in Twemproxy. We do not need this license anymore.

File: src/dyn_types.c [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#include "dyn_types.h"
```
Comment (ipapapa) [inline src/dyn_types.c]: Are you planning to expand `dyn_types.c`?

File: src/dyn_message.c [hunk 3]
```diff
@@ -162,17 +162,21 @@ static struct string msg_type_strings[] = {
...
     return fprintf(stream, "<REQ %p %lu:%lu %.*s>", req, req->id, req->parent_id,
                    req_type->len, req_type->data);
 }
 
-int
-print_rsp(FILE *stream, struct msg *rsp)
+static int
+print_rsp(FILE *stream, const struct object *obj)
 {
+    ASSERT(obj->type == OBJ_REQ);
```
Comment (ipapapa) [inline src/dyn_message.c]: It is nice to add the `ASSERT` but are they actually working? In the past debug had some issues.

Comment (shailesh33) [inline src/dyn_message.c]: asserts are working with the new build process that we did in our jenkins job. but I will check

File: src/dyn_types.c (hunk 2 above)
Comment (shailesh33) [inline src/dyn_types.c]: Depends. not sure.

File: src/dyn_connection_internal.c (hunk 1 above)
Comment (shailesh33) [inline src/dyn_connection_internal.c]: ok

File: src/dyn_types.c (hunk 2 above)
Comment (ipapapa) [inline src/dyn_types.c]: then why did you create a new file just for migrating a small function?

Comment (shailesh33) [inline src/dyn_types.c]: yes for now. I didnt want to put in dyn_core but I will revisit it.

**Pull request merged**